### PR TITLE
platform: update stm32mp1 reference

### PR DESCRIPTION
**Pull request by etienne-lms**
Add URL for stm32mp1 platforms information and update availability since the platform is now publicly available.

### CODE REVIEW

File: general/platforms.rst [hunk 1]
```diff
@@ -213,9 +213,9 @@ please refer to the file MAINTAINERS_ for contact details for various platforms.
      - No
      - Yes
 
-   * - STMicroelectronics stm32mp1
+   * - `STMicroelectronics stm32mp1 series - <http://www.st.com/stm32mp1>`
```
Comment (jbech-linaro) [inline general/platforms.rst]: Seems like that on the ST site it's written in upper case (STM32MP1). Overwise ...
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

Comment (etienne-lms) [reply general/platforms.rst]: You're right. Thanks. I'll fix.

**Comment by etienne-lms** [comment]
Tag applied, thanks.

**Comment by jbech-linaro** [comment]
Manually pushed (with tags), closing PR.

**Pull request closed without merging**